include/boost/hana/config.hpp

```diff
@@ -70,7 +70,7 @@ Distributed under the Boost Software License, Version 1.0.
 //////////////////////////////////////////////////////////////////////////////
 // Check the compiler for general C++14 capabilities
 //////////////////////////////////////////////////////////////////////////////
-#if (__cplusplus < 201400)
+#if (__cplusplus < 201400) && (defined(__clang__) || !defined(_MSC_VER))
```

Comment:
> Why not simply #if (__cplusplus < 201400) && !defined(_MSC_VER) ?

Because I didn't know that clang in MSVC mode also supports `#pragma message`. But it does, just tested with clang/c2, so I've updated to use `#pragma message` in MSVC mode.